src/types.ts

```diff
@@ -101,3 +101,45 @@ export interface Column {
   stylingInfo: string[];
   sortDirection: string;
 }
+
+export type TableData =
+  | {
+      totalColumnCount: number;
+      totalRowCount: number;
+      totalPages: number;
+      paginationNeeded: boolean;
+      rows: number[];
+      columns: number[];
+      totalsPosition: string;
+    }
+  | null
```

Comment:
when is this null?

Reply:
I realized that the type TableData is not needed in this PR. Gonna remove it.